public/app/plugins/datasource/loki/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -138,6 +150,22 @@ const MonacoQueryField = ({ history, onBlur, onRunQuery, initialValue, datasourc
     editor.onDidChangeModelContent(checkDecorators);
   };
 
+  const onType = debounce(async (query: string) => {
+    if (isValidQuery(query) === false) {
+      return;
+    }
+
+    const timerange = datasource.getTimeRange();
+    const previousTimerange = undefined;
+    const previousQuery = undefined;
+
+    const update = shouldUpdateStats(query, previousQuery, timerange, previousTimerange);
+    if (update) {
+      const stats = await makeStatsRequest(datasource, query);
+      stats ? setQueryStats!(stats) : setQueryStats!(undefined);
```

Comment:
`stats` here can be only stats or undefined, so we could just do `setQueryStats!(stats)`